### ISSUE TITLE
Use of uninitialized value in UR/BoolExpr/Util.pm

### ISSUE THREAD

**Issue by obigriffith** [genome/gms#80]
Multiple steps in Clinseq are also now stuck on this:

2014-01-21 19:20:14-0600 GGM: Use of uninitialized value in numeric ne (!=) at /opt/gms/AUIS907/sw/ur/lib/UR/BoolExpr/Util.pm line 65.
2014-01-21 19:20:15-0600 GGM: 
2014-01-21 19:20:15-0600 GGM: 
2014-01-21 19:20:15-0600 GGM: Writing IGV XML session files for clinseq build 34482e23408345dc810f02b76e7e16b9 to: /opt/gms/AUIS907/fs/AUIS907/info/model_data/9139d1d4d0d6481db98962e328c724ee/build34482e23408345dc810f02b76e7e16b9/TST1/igv/
2014-01-21 19:20:16-0600 GGM: Use of inherited AUTOLOAD for non-method Genome::Utility::List::join_with_single_slash() is deprecated at /opt/gms/AUIS907/sw/genome/lib/perl/Genome/Model/ClinSeq/Command/DumpIgvXml.pm line 471.
2014-01-21 19:20:16-0600 GGM: Command module died or returned undef.
2014-01-21 19:20:16-0600 GGM: Can't locate object method "join_with_single_slash" via package "Genome::Utility::List" at /opt/gms/AUIS907/sw/workflow/lib/Workflow/Server/Worker.pm line 132
2014-01-21 19:20:16-0600 GGM:   Genome::Model::ClinSeq::Command::DumpIgvXml::generate_track_xml('Genome::Model::ClinSeq::Command::DumpIgvXml=HASH(0x57ac060)', '-build', 'Genome::Model::Build::ReferenceAlignment::Solexa=HASH(0x7993398)', '-resource_type', 'bam') called at /opt/gms/AUIS907/sw/genome/lib/perl/Genome/Model/ClinSeq/Command/DumpIgvXml.pm line 194
2014-01-21 19:20:16-0600 GGM:   Genome::Model::ClinSeq::Command::DumpIgvXml::execute('Genome::Model::ClinSeq::Command::DumpIgvXml=HASH(0x57ac060)') called at /opt/gms/AUIS907/sw/ur/lib/Command/V2.pm line 214
2014-01-21 19:20:16-0600 GGM:   Command::V2::execute('Genome::Model::ClinSeq::Command::DumpIgvXml=HASH(0x57ac060)') called at /opt/gms/AUIS907/sw/workflow/lib/Workflow/OperationType/Command.pm line 362

**Comment by obigriffith** [comment]
This error occurs in the following steps:
Creating a summary of input builds using summarize-builds
Create IGV XML session files for varying levels of detail using the input builds
etc...

**Comment by malachig** [comment]
Here is the problematic line:

``` perl
  #Set the resource file url path
  $resource_file_url = Genome::Utility::List::join_with_single_slash($ENV{GENOME_SYS_SERVICES_FILES_URL}, $resource_file);
```

I expect that $GENOME_SYS_SERVICES_FILES_URL was recently added as part of ongoing efforts to make the whole system more configurable.  This environment variable is new and therefore has not been defined in the sGMS.

Perhaps we should define this value in /etc/genome.conf in the sGMS.

Within TGI it is set to 'https://gscweb.gsc.wustl.edu/'.  We could set it to be 'localhost' initially since that is where our web service lives.

**Comment by obigriffith** [comment]
I only see "Genome::Utility::List::join_with_single_slash" in the master branch, not gms-pub.

**Comment by gatoravi** [comment]
The Genome::Utility::List::join_with_single_slash() subroutine was added by @amb43790 on Nov 22(before the merge) to use the environment variable  $GENOME_SYS_SERVICES_FILES_URL genome/gms-core@c4f25210e0970e27b4f5b3dd529d286dd9c9ffc7. We will have to decide on a way to incorporate this change on the standalone side. As @malachig mentioned, we could set this variable to 'localhost' and then merge the Genome::Utility::List file with master. 

Note - This variable is being used in this case to set the base url for IGV sessions.

**Comment by sakoht** [comment]
Does that method exist in Genome::Utility::List?

If not it should fail in production as well.

**Comment by sakoht** [comment]
How did we get part of gms-pub using that code, while the parts that define it are not there?

Is this a case of code manually brought over to gms-pub?

**Comment by gatoravi** [comment]
yep, we decided to use the latest version of Clinseq so we manually brought that over to gms-pub.

**Comment by obigriffith** [comment]
It looks like Avi copied over the the missing method 'join_with_single_slash' in Genome::Utility::List from master to gms-pub and the crashed steps seem to succeed now.

**Comment by gatoravi** [comment]
this has been fixed in genome/gms-core@71f053b1fc74ad0860c27ab9335ccf459253cd75 and genome/gms-core@8937d55acd
